lib/grammers-tl-gen/src/rustifier.rs

```diff
@@ -98,20 +98,20 @@ pub mod definitions {
         };
 
         match variant {
-            "" | _ if variant.chars().all(char::is_numeric) => {
-                // Use the name from the last uppercase letter
+            "Self" => {
+                // Use the name from the second-to-last uppercase letter
                 &name[name
                     .as_bytes()
                     .iter()
+                    .take(name.len() - variant.len())
                     .rposition(|c| c.is_ascii_uppercase())
                     .unwrap_or(0)..]
             }
-            "Self" => {
-                // Use the name from the second-to-last uppercase letter
+            _ if variant.chars().all(char::is_numeric) => {
```

Comment:
Curious to know why this got re-ordered.

Reply:
Clippy warn the case `"" | _` so I changed the order of match the cases and used only `_`

Reply:
I remember now, I left the explicit `""` arm because it's not obvious that we also want to match empty strings with `all`. Can we add a comment?

```rust
// Note: this intentionally handles the empty string case too.
```

Reply:
Done!